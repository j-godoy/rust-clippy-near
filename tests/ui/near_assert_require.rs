#![warn(clippy::near_assert_require)]

fn main() {
    // test code goes here
    let x = 10;
    // assert!(x < 10 * 8);
    // foo()
}

pub struct StatusMessage {
    // records: HashMap<AccountId, String>,
}

impl StatusMessage {
    pub fn foo() {
        let y= 11;
        assert!(y < 10 * 8);
    }
}
