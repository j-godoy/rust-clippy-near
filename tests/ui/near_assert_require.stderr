error: used `assert!` from standard rust
  --> $DIR/near_assert_require.rs:17:9
   |
LL |         assert!(y < 10 * 8);
   |         ^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::near-assert-require` implied by `-D warnings`
note: use `require` for near contracts
  --> $DIR/near_assert_require.rs:17:9
   |
LL |         assert!(y < 10 * 8);
   |         ^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

